src/main/java/com/hivemq/cli/commands/shell/ListClientsCommand.java

```diff
@@ -178,18 +175,19 @@ public void run() {
     }
 
     private Map<String, String> mapKeyToPrettyOuput(final @NotNull String[] sortedKeys) {
-        final ClientCache<String, MqttClient> cache = mqttClientExecutor.getClientCache();
+        final Map<String, ClientData> clientKeyToClientData = MqttClientExecutor.getClientDataMap();
         final Map<String, String> keyToPretty = new HashMap<>();
+
         for (int i = 0; i < sortedKeys.length; i++) {
-            MqttClient client = cache.get(sortedKeys[i]);
+            final MqttClient client = clientKeyToClientData.get(sortedKeys[i]).getClient();
             keyToPretty.put(sortedKeys[i], client.getConfig().getClientIdentifier().get() + "@" + client.getConfig().getServerHost());
         }
         return keyToPretty;
     }
 
     public String[] getSortedClientKeys() {
-        final Set<String> keys = mqttClientExecutor.getClientCache().keySet();
-        final Map<String, ClientData> clientDataMap = mqttClientExecutor.getClientDataMap();
+        final Set<String> keys = MqttClientExecutor.getClientDataMap().keySet();
```

Comment:
This method should return a sorted List of ClientData and should not work on the keys which are then used to retrieve the ClientData from the Map again.